Source/DafnyCore/Options/CommonOptionBag.cs

```diff
@@ -31,6 +31,10 @@ public class CommonOptionBag {
     "Print additional information such as which files are emitted where.") {
   };
 
+  public static readonly Option<bool> WarnDeprecation = new("--warn-deprecation", () => true,
```

Comment:
We need a general mechanism for turning off warnings, and once we have that options like these become obsolete and should be removed, so maybe it's better not to add them in the first place.